pkg/util/informermanager/single-cluster-manager.go

```diff
@@ -98,5 +112,8 @@ func (s *singleClusterInformerManagerImpl) Start(stopCh <-chan struct{}) {
 }
 
 func (s *singleClusterInformerManagerImpl) WaitForCacheSync(stopCh <-chan struct{}) map[schema.GroupVersionResource]bool {
-	return s.informerFactory.WaitForCacheSync(stopCh)
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.syncedInformers = s.informerFactory.WaitForCacheSync(stopCh)
```

Comment:
The `syncedInformers` will be overridden every time  `WaitForCacheSync` called, a little bit expensive.

Reply:
Yes, but this function will not called every time because the check function `IsInformerSynced` and `IsHandlerExist` are called before. If the informer has synced, `WaitForCacheSync` will not be called. I think this expense is worth it.